src/framework/theme/components/select/select.component.ts

```diff
@@ -385,7 +381,7 @@ export class NbSelectComponent<T> implements OnInit, AfterViewInit, AfterContent
    * Selects option or clear all selected options if value is null.
    * */
   protected handleSelect(option: NbOptionComponent<T>) {
-    if (option.value) {
+    if (this.options.some((opt) => opt.content === option.content)) {
```

Comment:
@nnixaa @Tibing I really don't like thi but couldn't find a better way. Basically, in order to know if we want to update the selection we search the existing options and try to match based on the `nb-option` content (which is its label). if you guys have a better idea I'm definitely opened.

PS: we could expose besides the `content` property a new `value` property which would tap into the native element's attributes and take the value from there, but we'd still have to do this kind of check in order to bypass javascript's falsy values

Reply:
Probably we could use `option.value != null`. Loose comparison with `null` is true only for `null` and `undefined`, so it'll bypass any other falsy values.
@aefox Do you see any problems with this solution?

Could you also add small clarification to the select docs, please? Currently, it says 'Items without values will clean selection.' We also should mention `null` and `undefined`.

And this is a breaking change since behavior for people using falsy values (except `null` and `undefined`) will change.

Reply:
@yggg thanks a lot for the review!
Indeed you are correct, loose comparison to null is definitely the way to go here, nicely spotted. Also, nice catch on the docs, updated those as well.

Not sure about the breaking changes. Should I do anything in particular for this?

Reply:
>Not sure about the breaking changes. Should I do anything in particular for this?

Commit which introduce breaking change should has `BREAKING CHANGE:` with description inside commit message body. See details in [CONTRIBUTING.md](https://github.com/akveo/nebular/blob/master/CONTRIBUTING.md#footer). Please, update commit and force push.

It also means we would be able to merge this PR only in major version update (Nebular 4.x.x) since we follow the [semantic versioning spec](https://semver.org/).